vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java

```diff
@@ -325,7 +338,280 @@ public void queryHostMetaData(SDDCSoftwareConfig vc) {
+      
+      MetricId[] queryAvailableMetric =
+            performanceManager.queryAvailableMetric(hostRef, null, null, new Integer(20));
+      
+      List<MetricId> metricIdList = new ArrayList<MetricId>();
+      if (queryAvailableMetric.length > 0) {
+         for (int i = 0; i < queryAvailableMetric.length; i++) {
+            MetricId metricId = queryAvailableMetric[i];
+            int counterId = metricId.getCounterId();
+            String instanceId = metricId.getInstance();
+            if (counters.containsKey(new Integer(counterId)) && (instanceId == null || instanceId.isEmpty())) {
+               metricIdList.add(metricId);
+            }
+         }
+      }
+
+      return metricIdList;
+   }
+   
+   public void feedHostUsageData(VsphereClient vsphereClient, String assetId,
+         ManagedObjectReference hostRef) {
+
+      List<RealTimeData> realTimeDatas = new ArrayList<RealTimeData>();
+      RealTimeData realTimeData = new RealTimeData();
+      realTimeData.setAssetID(assetId);
+      List<ValueUnit> valueUnits = new ArrayList<ValueUnit>();
+ 
+      PerformanceManager performanceManager = vsphereClient.getPerformanceManager();
+
+      Map<Integer, String> counters = getMetricsCounters(performanceManager);
+      if(counters == null || counters.isEmpty()) {
+         logger.error("Asset: {} failed to get performance counters", assetId);
+         return;
+      }
+      List<MetricId> metricIdList = getPerformenceMetricsIds(performanceManager, hostRef, counters);
+      if(metricIdList == null || metricIdList.isEmpty()) {
+         logger.error("Asset: {} failed to get performance metricIds", assetId);
+         return;
+      }
+      //fill spec info
+      ProviderSummary summary = performanceManager.queryProviderSummary(hostRef);
+      int perfInterval = summary.getRefreshRate();
+      QuerySpec[] specs = new QuerySpecImpl[1];
+      specs[0] = new QuerySpecImpl();
+      specs[0].setEntity(hostRef);
+      specs[0].setMetricId(metricIdList.toArray(new MetricId[metricIdList.size()]));
+      specs[0].setIntervalId(perfInterval);
+      specs[0].setMaxSample(15);
+
+      //get metrics value
+      EntityMetricBase[] metricBase = performanceManager.queryStats(specs);
+      if(metricBase == null || metricBase.length <= 0) {
+         logger.error("Asset: {} failed to get performance metricBase", assetId);
+         return;
+      }
+      for(EntityMetricBase entityMetricBase : metricBase) {
+         EntityMetric entityMetric = (EntityMetric) entityMetricBase;
+         MetricSeries[] metricSeries = entityMetric.getValue();
+         SampleInfo[] sampleInfos = entityMetric.getSampleInfo();
+         for (MetricSeries metricSerie : metricSeries) {
+            if (metricSerie instanceof IntSeries) {
+               IntSeries intSeries = (IntSeries) metricSerie;
+               long[] values = intSeries.getValue();
```

Comment:
please check what will happen if it don't have 15 data point.

Reply:
The size of the array must match the size of sampleInfo property in the PerfEntityMetric that contains this series.